src/XtensaOptimize.cpp

```diff
@@ -132,6 +97,16 @@ Type get_native_xtensa_vector(const Type &t, const Target &target) {
     return t.with_lanes(vector_bitwidth / t.bits());
 }
 
+bool is_native_vector_type(const Type &t, const Target &target) {
+    Type native_vector_type = get_native_xtensa_vector(t, target);
+    return t == native_vector_type;
+}
+
+bool is_double_native_vector_type(const Type &t, const Target &target) {
+    Type native_vector_type = get_native_xtensa_vector(t, target);
+    return t == native_vector_type.with_lanes(native_vector_type.lanes());
```

Comment:
Should there be a `* 2` inside the `with_lanes()` call?

Reply:
Thanks, fixed.